sol/thinf/f21t1.tex

```diff
@@ -0,0 +1,189 @@
+		(q_24) edge node {$b$} (q_err)
+		(q_err) edge [loop below] node {$a,b$} (q_err);
+\end{tikzpicture}
+\end{center}
+\par
+
+\subsubsection{Aufgabe 3}
+
+Die Folge $1,1,1,1$ ist keine Lösung von $K$: Das Wort $x$ ist länger als die
+Wörter $y$ oder $z$, und $y$ und $z$ unterscheiden sich im ersten Buchstaben.
+
+Die Folge $2,2,2,2$ ist eine Lösung von $K$, da dann $y = z$ gilt.
+
+PCP lässt sich auf DPCP wie folgt reduzieren:
+\[
+	f(K) :=
+	\begin{cases}
+		(x_1,y_1,1), \ldots, (x_k,y_k,k) & \text{falls } K = (x_1,y_1), \ldots, (x_k,y_k) \\
+		\epsilon & \text{sonst}
+	\end{cases}
+\]
+Dabei sei o.B.d.A. angenommen, dass $\Sigma \cap \Nat = \emptyset$.
+
+\paragraph{$f$ ist total} Da $f$ für jede Eingabe definiert ist, ist $f$ total.
+\par
+
+\paragraph{$f$ ist berechenbar} Ein Syntaxcheck zu Beginn ist möglich. Das
+Ergänzen eines Tupels ist in $O(1)$ möglich, also kann der neue Ausdruck in
+$O(k)$ niedergeschrieben werden.
+\par
+
+\paragraph{$f$ ist korrekt}
+\begin{align*}
+	K \in PCP &\iff K = \ldots \land \exists m.\, x = y \\
+		  &\iff f(K) = \ldots \land \exists m.\, x = y\\
+		  &\iff f(K) = \ldots \land \exists m.\, x = y \lor x = z \lor y
+		  = z \\
+		  &\iff f(K) \in DPCP
+\end{align*}
+
+\subsubsection{Aufgabe 4}
+
+\paragraph{Umwandlung von Klauseln mit 0 Literalen}
+Diese Klauseln evaluieren zu $0$. Folglich werden die folgenden Klauseln
+generiert:
+\[
+	\left\{\{x,y,z\} \mid x \in \{a,\lnot a\}, y \in \{b,\lnot b\}, z \in
+	\{c,\lnot b\}\right\}
+\]
+Diese Klauseln können nicht alle gemeinsam erfüllt werden.
+\par
+\paragraph{1 Literal}
+Sei dieses eine Literal $x$. Generiere folgende Klauseln:
+\[
+	\left\{\{x,y,z\} \mid y \in \{b,\lnot b\}, z \in
+	\{c,\lnot b\}\right\}
+\]
+\par
+\paragraph{2 Literale}
+Seien die beiden Literale $x$ und $y$. Generiere folgende Klauseln:
+\[
+	\left\{\{x,y,z\} \mid z \in
+	\{c,\lnot b\}\right\}
+\]
+\par
+\paragraph{3 Literale}
+Kopiere die Klausel in die gerade zu generierende Genau-3-SAT-Formel.
+\par
+
+Die Generierung der neuen Klauseln ist pro Klausel in $O(1)$ schaffbar, damit
+hängt die Komplexität der Umwandlung linear von der Anzahl der Klauseln ab, ist
+also in polynomieller Zeit berechenbar. Sie ist auch offensichtlich total und
+korrekt.
+
+Da $3-CNF-SAT$ zu $GENAU-3-CNF-SAT$ reduziert, ist letzteres Problem mindestens
+so schwer wie $3-CNF-SAT$. Da $3-CNF-SAT$ NP-vollständig, also insbesondere
+NP-hart ist, folgt, dass $3-CNF-SAT$ und damit auch $GENAU-3-CNF-SAT$ mindestens
+in NP liegt. Es fehlt also nur noch der Beweis, dass $GENAU-3-CNF-SAT$
+nicht-deterministisch in polynomieller Zeit entscheidbar ist. Ein entsprechender
+Algorithmus verhält sich wie folgt:
+\begin{enumerate}
+	\item
+		Wähle nicht-deterministisch eine erfüllende Belegung $\kappa$.
+	\item
+		Überprüfe für jede der $k$ Klauseln, ob eines der drei Literale
+		von $\kappa$ erfüllt wird. ($O(k)$)
+	\item
+		Falls das für alle Klauseln gilt, akzeptiere, ansonsten
+		verweigere.
```

Comment:
Ist verweigern in diesem Kontext ein gängiger Begriff? Habe es zumindest noch nie gehört und wäre jetzt bei "akzeptiere nicht" geblieben.

Reply:
done